pkg/comp-functions/functions/common/maintenance/maintenance_test.go

```diff
@@ -177,6 +177,68 @@ func TestAddMaintenanceJob(t *testing.T) {
 	}
 }
 
+func TestInitialMaintenanceJob(t *testing.T) {
+	tests := []struct {
+		name                      string
+		initialMaintenanceRan     bool
+		wantInitialJob            bool
+		wantInitialMaintenanceRan bool
+		fileName                  string
+	}{
+		{
+			name:                      "GivenNewInstance_ThenExpectInitialJob",
+			initialMaintenanceRan:     false,
+			wantInitialJob:            true,
+			wantInitialMaintenanceRan: true,
+			fileName:                  "vshn-postgres/maintenance/01-GivenSchedule.yaml",
+		},
+		{
+			name:                      "GivenInitialMaintenanceAlreadyRan_ThenExpectNoJob",
+			initialMaintenanceRan:     true,
+			wantInitialJob:            false,
+			wantInitialMaintenanceRan: true,
+			fileName:                  "vshn-postgres/maintenance/01-GivenSchedule.yaml",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.TODO()
+
+			svc := commontest.LoadRuntimeFromFile(t, tt.fileName)
+
+			comp := &vshnv1.VSHNPostgreSQL{}
```

Comment:
PostgreSQL is the only non-helm based service. We should probably also test with a helm based one.